CONTRIBUTING.md

```diff
@@ -24,6 +24,234 @@ This project also supports using
 [Visual Studio Code](https://code.visualstudio.com). Install the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp) and install the [.NET Core CLI](https://get.dot.net/core) to get started.
 
 
+## Guidelines
+
+When preparing to submit changes to this project, please guidelines.
+
+### Practice Behavior Driven Development (BDD)
+
+![Diagram showing the interaction between the BDD and TDD process. A failing BDD scenario is an input to the TDD process. The TDD process is to Write a test, make the test pass, refactor. If the scenario is still failing, then the TDD process repeats. Once the scenario passes, then the BDD process repeats.](https://cucumber.io/cucumber/media/images/blog/bdd-tdd-loop-labelled.png)
+> Diagram source: [_BDD builds momentum_](https://cucumber.io/blog/bdd/bdd-builds-momentum/) blog post by Seb Rose
+
+Start all of your submissions with a behavior specification. We sometimes call these acceptance tests, because they help us verify that the `freshli` application behaves in an acceptable way from the point of view of a customer.
+
+BDD is a customer-centric process. It helps ensure that we're describing the expected behavior of the `freshli` application from the perspective of the human that will be using it. Additionally, by taking this approach, we're able to have a hirer degree of confidence that the application behaves the way that we expect it to.
```

Comment:
I like this way of thinking, and phrasing it in the documentation. I ended on the acceptance test because that's what I always did. Acceptance tests for me are expensive to run so I tend to avoid them until I'm confirming on what goes outside is working okay as well. By turning it around I think you make a valid case that the focus is customer centric.